development/components/database/objectmodel.md

```diff
@@ -64,86 +104,443 @@ public static $definition = [
+
+### Hard or soft delete an object
+
+Two delete mecanisms are available with ObjectModel : hard delete and soft delete. 
+Hard-delete deletes the record from the database, while soft-delete sets a flag to database indicating that this record is deleted.
+
+{{% notice info %}}
+**Soft delete is not always available.**
+
+If the model object has no `deleted` property or no `deleted` definition field, a `PrestaShopException` will be thrown
+{{% /notice %}}
+
+{{% notice info %}}
+**Soft delete does not trigger DeleteHooks.**
+
+Soft deleting an object does not trigger **Delete** related hooks, but will trigger **Update** related hooks. [ObjectModel lifecycle hooks]({{< ref "#lifecycle-hooks" >}})
+{{% /notice %}}
 
 ```php
-'multishop' => true
+$id = 2; // id of the object in database
+$cms = new Cms($id); 
+$cms->softDelete(); // sets the deleted property to true, and triggers an update() call
+...
+$cms->delete(); // triggers a DELETE statement to the DBAL
 ```
 
-In order to retrieve an object which depends on the current store, and in many languages:
+## Advanced usage 
+
+### Multiple languages objects{#multiple-languages}
+
+PrestaShop's ObjectModel can handle translations (also called internationalization, or i18n) of your objects.
+
+#### Under the hood : how does it work ?
+
+When declaring a multi-language ObjectModel, PrestaShop will fetch another database table named like your base database table, with a suffix `_lang`
+This table references the id of the base Object (`id_cms`), the id of the language (`id_lang`), and each translatable field. 
+
+In our previous example, for `Cms` ObjectModel :
+
+<div class='mermaid'>
+classDiagram
+    ps_cms <|-- ps_cms_lang
+    ps_cms : id_cms
+    ps_cms : id_cms_category
+    ps_cms : position
+    ps_cms : active
+    ps_cms : ...
+    class ps_cms_lang {
+        id_cms
+        id_lang
+        meta_title
+        meta_description
+        meta_keywords
+        link_rewrite
+        content
+        ...
+    }
+</div>
+
+#### Translate your ObjectModel entity
+
+To do so, you must declare the `multilang` setting of your model definition to `true` :
 
 ```php
-'multilang_shop' => true
+public static $definition = [
+    ...
+    'multilang' => true,
+    ...
 ```
 
-## Main methods
+And then, you must declare which ones of your fields are translated :
 
+```php
+'fields' => array(
+    ...
+    'meta_description' => [
+        ...
+        'lang' => true,
+        ...
+    ],
+    ...
+)
+```
 
-{{% funcdef %}}
+#### Accessors for Translatable ObjectModels{#multiple-language-accessors}
 
-__construct($id = NULL, $id_lang = NULL)
-: 
-    Build object.
+Translatable fields are available in your ObjectModel as `array`. 
+In our example, to update the attributes `meta_title` for languages EN (`$lang_id=1`) and FR (`$lang_id=2`), use the following method :
```

Comment:
```suggestion
In our example, to update the attributes `meta_title` for languages EN (`$lang_id=1`) and FR (`$lang_id=2`), use the following method:
```

Reply:
maybe it is somewhere further, but I don't see any mention that if you pass `$idLang` to the constructor you'll not have an array in the multi-lang properties